packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -442,10 +442,7 @@ export default class {
 
     const xapi = this._xapis[id]
     delete this._xapis[id]
```

Comment:
Shouldn't this line also be removed?

Reply:
I think so: https://github.com/vatesfr/xen-orchestra/blob/b62917edf42c00aeb039879b448ca93bf2e4b678/packages/xo-server/src/xo-mixins/xen-servers.js#L426

Reply:
No, because we can disconnect a connecting server. See https://github.com/vatesfr/xen-orchestra/pull/4684/files#diff-9b31392ad9c6d4090e97f3073e0d7033L312

Reply:
what about deleting the `xapi` entry in the bloc you pointed and removing the code in `disconnectXenServer`?

Reply:
Deleting the entries mean that the user want to disconnect his server. See https://github.com/vatesfr/xen-orchestra/pull/4684/files#diff-9b31392ad9c6d4090e97f3073e0d7033L29

Reply:
Nevertheless, I think we should centralize this in one place to make sure everything is correctly handled.

And I'm wondering if that's not already the case in `connectXenServer` (the `catch` clause). If that's the case it's not obvious and we should add some comments to make it clearer 🙂

Reply:
How we'll know that the user would disconnect his server. It have already a commend in `connectXenServer`  describing this case. See https://github.com/vatesfr/xen-orchestra/pull/4684/files#diff-9b31392ad9c6d4090e97f3073e0d7033R311